draft-ietf-quic-http.md

```diff
@@ -1042,17 +1045,89 @@ The GOAWAY frame applies to the connection, not a specific stream.  An endpoint
+type HTTP_WRONG_STREAM.
+
+A server MUST NOT send a MAX_PUSH_ID frame.  A client MUST treat the receipt of
+a MAX_PUSH_ID frame as a connection error of type HTTP_MALFORMED_FRAME.
+
+The maximum Push ID is unset when a connection is created, meaning that a server
+cannot push until it receives a MAX_PUSH_ID frame.  A client that wishes to
+manage the number of promised server pushes can increase the maximum Push ID by
+sending a MAX_PUSH_ID frame as the server fulfills or cancels server pushes.
+
+~~~~~~~~~~  drawing
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          Push ID (i)                        ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~~~~~~~~
+{: #fig-max-push title="MAX_PUSH_ID frame payload"}
+
+The MAX_PUSH_ID frame carries a single variable-length integer that identifies
+the maximum value for a Push ID that the server can use (see
+{{frame-push-promise}}).  A MAX_PUSH_ID frame cannot reduce the maximum Push ID;
+receipt of a MAX_PUSH_ID that contains a smaller value than previously received
+MUST be treated as a connection error of type HTTP_MALFORMED_FRAME.
+
+A server MUST treat a MAX_PUSH_ID frame payload that does not contain a single
+variable-length integer as a connection error of type
+HTTP_MALFORMED_FRAME.
+
+
+# Connection Closure
+
+Once established, an HTTP/QUIC connection can be used for many requests and
+responses over time until the connection is closed.  Connection closure can
+happen in any of several different ways.
+
+## Idle Connections
+
+Each QUIC endpoint declares an idle timeout during the handshake.  If the
+connection remains idle (no packets received) for longer than this duration, the
+peer will assume that the connection has been closed.  HTTP/QUIC implementations
+will need to open a new connection for new requests if the existing connection
+has been idle for longer than the server's advertised idle timeout, and SHOULD
+do so if approaching the idle timeout.
+
+HTTP clients are expected to use QUIC PING frames to keep connections open while
+there are responses outstanding for requests or server pushes. If the client is
+not expecting a response from the server, allowing an idle connection to time
+out is preferred over expending effort maintaining a connection that might not
+be needed.  A gateway MAY use PING to maintain connections in anticipation of
+need rather than incur the latency cost of connection establishment to servers.
+Servers SHOULD NOT use PING frames to keep a connection open.
+
+## Connection Shutdown
+
+Even when a connection is not idle, either endpoint can decide to stop using the
+connection.  Clients do this by not sending additional requests on the
+connection; responses and pushed responses associated to previous requests will
+continue to completion.
```

Comment:
This introduction doesn't mention that the intent is to be "graceful" or to close the connection without losing requests.  That's important, because it is also possible to just slam the connection closed.